edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/twin/TwinStoreEntity.cs

```diff
@@ -27,7 +27,7 @@ public TwinStoreEntity(TwinCollection reportedPropertiesPatch)
         public TwinStoreEntity(Twin twin, TwinCollection reportedPropertiesPatch)
         {
             this.Twin = Option.Maybe(twin);
-            this.ReportedPropertiesPatch = reportedPropertiesPatch?.Count != 0
+            this.ReportedPropertiesPatch = reportedPropertiesPatch != null && reportedPropertiesPatch.Count != 0
```

Comment:
Do you have tests that cover this logic?

Reply:
just added tests